src/main/groovy/org/spongepowered/gradle/ore/OreDeployTask.groovy

```diff
@@ -0,0 +1,151 @@
+        }
+
+        logger.quiet('Publishing ' + plugin.name + ' to ' + instanceUrl + '.')
+        logger.quiet('  Recommended: ' + recommended)
+        logger.quiet('  Channel: ' + channel)
+
+        if (instanceUrl.endsWith("/")) {
+            instanceUrl = instanceUrl.substring(0, instanceUrl.length() - 1)
+        }
+        URL projectUrl
+        try {
+            String pluginId = project.sponge.plugin.id
+            String str = instanceUrl + '/api/projects/' + pluginId + '/versions/' + project.version
+            projectUrl = new URL(str)
+            logger.debug('POST ' + str)
+        } catch (MalformedURLException e) {
+            throw new InvalidUserDataException("Invalid project URL", e)
+        }
+
+        if (apiKey == null) {
+            apiKey = project.property('oreDeploy.apiKey')
+        }
+        if (channel == null) {
+            throw new InvalidUserDataException("Missing channel name.")
+        }
+        HttpEntity requestEntity = MultipartEntityBuilder.create()
+            .addPart('apiKey', new StringBody(apiKey))
+            .addPart('channel', new StringBody(channel))
+            .addPart('recommended', new StringBody(recommended as String))
+            .addPart('pluginFile', new FileBody(plugin.file))
+            .addPart('pluginSig', new FileBody(sig.file))
+            .build()
+        logger.debug(requestEntity.toString())
+
+        HttpClient http = HttpClients.createDefault()
+        HttpPost post = new HttpPost(projectUrl.toURI())
+        post.entity = requestEntity
+        HttpResponse response = null
+        try {
+            response = http.execute(post)
+            HttpEntity responseEntity = response.entity
+            def status = response.statusLine
+            boolean created = status.statusCode == HttpURLConnection.HTTP_CREATED
+            if (!created) {
+                logger.error('[failure] ' + status.statusCode + ' ' + status.reasonPhrase)
+                printErrorStream responseEntity.content
+            } else {
+                def json = new JsonSlurper().parse(responseEntity.content, 'UTF-8')
+                logger.debug(json.toString())
+                logger.quiet('[success] ' + instanceUrl + (json.href as String))
+            }
+            EntityUtils.consume(responseEntity)
+            if (!created) {
+                throw new GradleException('Deployment failed.')
+            }
+        } catch (InterruptedIOException | SocketException e) {
+            throw new GradleException('Failed to connect to Ore.', e)
+        } catch (SSLHandshakeException ignored) {
+            throw new GradleException(
+                'Please update to Java version 1.8.0_121+ in order to connect to Sponge securely.')
+        } catch (IOException e) {
+            throw new IOException('An unexpected error occurred.', e)
+        } finally {
+            if (response != null) {
+                response.close()
```

Comment:
Groovy has `.withCloseable` similar to `.withStream` that does this automatically